@here/olp-sdk-core/lib/utils/MultiPartUpload.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (C) 2021 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+import {
+    DataStoreRequestBuilder,
+    HRN,
+    OlpClientSettings,
+    RequestFactory
+} from "@here/olp-sdk-core";
+import { BlobApi, ObjectStoreApi } from "@here/olp-sdk-dataservice-api";
+
+interface MultiPartUploadParams {
+    /**
+     * The instance of OlpClientSettings.
+     */
+    settings: OlpClientSettings;
+    /**
+     * The version od Blob API to upload.
+     */
+    blobVersion: "v1" | "v2";
+    /**
+     * The Layer ID to upload.
+     */
+    layerId: string;
+    /**
+     * If you're uploading to Blob V1, set the datahandle
+     * and set the key for Blob V2.
+     */
+    keyOrDatahandle: string;
+    /**
+     * The Catalog HRN to upload.
+     */
+    catalogHrn: HRN;
```

Comment:
Move up. The order should be in my opinion: `catalogHrn`, `layerId` and then `handle`.
But the question is why is it HRN and not base URL received by the lookup API? Is this wrapper doing the lookup himself or is already injected by the above class, i.e. VersionedLayerClient?

Reply:
This wrapper doing the lookup himself and can be used directly by user (with @dataservice-api for uploading to blob v1 or v2), also can be injected as you can see in VersionedLayerClient.

Reply:
I will improve a bit this. Lookup should not be performed in wrapper if the wrapper is used in versionedLayerClient (already performed and can be reused)